pkg/sitewise/api/util.go

```diff
@@ -20,129 +20,184 @@ var (
 	result := query
-	if query.PropertyAlias != "" {
-		resp, err := client.DescribeTimeSeriesWithContext(ctx, &iotsitewise.DescribeTimeSeriesInput{
-			Alias: util.GetPropertyAlias(query.BaseQuery),
-		})
-		if err != nil {
-			return models.AssetPropertyValueQuery{}, err
-		}
-		if resp.AssetId != nil {
-			result.AssetIds = []string{*resp.AssetId}
-		} else {
-			// For disassociated streams with a propertyAlias
-			result.AssetIds = []string{}
+	result.AssetPropertyEntries = []models.AssetPropertyEntry{}
+	// There should only be a list of property aliases OR lists for assetIds and propertyIds
+	// Look up the assetId and propertyId for a property alias
+	if len(query.PropertyAliases) > 0 {
+		for _, propertyAlias := range query.PropertyAliases {
+			resp, err := client.DescribeTimeSeriesWithContext(ctx, &iotsitewise.DescribeTimeSeriesInput{
+				Alias: aws.String(propertyAlias),
+			})
+			if err != nil {
+				return models.AssetPropertyValueQuery{}, err
+			}
+			if resp.AssetId != nil && resp.PropertyId != nil {
+				result.AssetPropertyEntries = append(result.AssetPropertyEntries, models.AssetPropertyEntry{
+					AssetId:       *resp.AssetId,
+					PropertyId:    *resp.PropertyId,
+					PropertyAlias: propertyAlias,
+				})
+			} else {
+				// For disassociated streams with just a propertyAlias
+				result.AssetPropertyEntries = append(result.AssetPropertyEntries, models.AssetPropertyEntry{
+					PropertyAlias: propertyAlias,
+				})
+			}
 		}
-		if resp.PropertyId != nil {
-			result.PropertyId = *resp.PropertyId
-		} else {
-			// For disassociated streams without a propertyAlias
-			result.PropertyId = ""
+	} else {
+		for _, assetId := range query.AssetIds {
+			for _, propertyId := range query.PropertyIds {
+				result.AssetPropertyEntries = append(result.AssetPropertyEntries, models.AssetPropertyEntry{
+					AssetId:    assetId,
+					PropertyId: propertyId,
+				})
+			}
 		}
 	}
 	return result, nil
 }
 
-func getAssetId(query models.BaseQuery) *string {
+func getFirstAssetId(query models.BaseQuery) *string {
 	if len(query.AssetIds) == 0 {
 		return nil
 	}
 	return aws.String(query.AssetIds[0])
 }
 
-func getPropertyId(query models.BaseQuery) *string {
-	if query.PropertyId == "" {
+func getFirstPropertyId(query models.BaseQuery) *string {
+	if len(query.PropertyIds) == 0 {
 		return nil
 	}
-	return aws.String(query.PropertyId)
+	return aws.String(query.PropertyIds[0])
 }
 
-func getPropertyAlias(query models.BaseQuery) *string {
-	if query.PropertyAlias == "" {
+func getFirstPropertyAlias(query models.BaseQuery) *string {
+	if len(query.PropertyAliases) == 0 {
 		return nil
 	}
-	return aws.String(query.PropertyAlias)
+	return aws.String(query.PropertyAliases[0])
 }
 
 func filterAnomalyAssetIds(ctx context.Context, client client.SitewiseClient, query models.AssetPropertyValueQuery) ([]string, error) {
 	anomalyAssetIds := []string{}
 
 	switch {
```

Comment:
Can we refactor this to an if/else statement, switch seems unnecessary

Reply:
Yes, I might have missed some places that had this two case switch statement